reference/test/distributed/vector_kernels.cpp

```diff
@@ -0,0 +1,162 @@
+
+#include <algorithm>
+#include <memory>
+#include <vector>
+
+
+#include <gtest/gtest-typed-test.h>
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/base/matrix_data.hpp>
+
+
+#include "core/distributed/vector_kernels.hpp"
+#include "core/test/utils.hpp"
+
+
+namespace {
+
+using comm_index_type = gko::distributed::comm_index_type;
+
+
+template <typename ValueLocalGlobalIndexType>
+class Vector : public ::testing::Test {
+protected:
+    using value_type =
+        typename std::tuple_element<0, decltype(
+                                           ValueLocalGlobalIndexType())>::type;
+    using local_index_type =
+        typename std::tuple_element<1, decltype(
+                                           ValueLocalGlobalIndexType())>::type;
+    using global_index_type =
+        typename std::tuple_element<2, decltype(
+                                           ValueLocalGlobalIndexType())>::type;
+    using global_entry = gko::matrix_data_entry<value_type, global_index_type>;
+    using mtx = gko::matrix::Dense<value_type>;
+
+    Vector() : ref(gko::ReferenceExecutor::create()), mapping{ref} {}
+
+    void validate(
+        const gko::dim<2> size,
+        const gko::distributed::Partition<local_index_type, global_index_type>*
+            partition,
+        std::initializer_list<global_index_type> input_rows,
+        std::initializer_list<global_index_type> input_cols,
+        std::initializer_list<value_type> input_vals,
+        std::initializer_list<
+            std::initializer_list<std::initializer_list<value_type>>>
+            output_entries)
+    {
+        std::vector<std::initializer_list<std::initializer_list<value_type>>>
+            ref_outputs;
+        auto input = gko::device_matrix_data<value_type, global_index_type>{
+            ref, size, input_rows, input_cols, input_vals};
+        for (auto entry : output_entries) {
+            ref_outputs.emplace_back(entry);
+        }
+        for (comm_index_type part = 0; part < partition->get_num_parts();
+             ++part) {
+            auto num_rows =
+                static_cast<gko::size_type>(partition->get_part_size(part));
+            auto output = mtx::create(ref, gko::dim<2>{num_rows, size[1]});
+            output->fill(gko::zero<value_type>());
+
+            gko::kernels::reference::distributed_vector::build_local(
+                ref, input, partition, part, output.get());
+
+            GKO_ASSERT_MTX_NEAR(output, ref_outputs[part], 0);
+        }
+    }
+
+    std::shared_ptr<const gko::ReferenceExecutor> ref;
+    gko::Array<comm_index_type> mapping;
+};
+
+TYPED_TEST_SUITE(Vector, gko::test::ValueLocalGlobalIndexTypes);
+
+
+TYPED_TEST(Vector, BuildsLocalEmpty)
+{
+    using local_index_type = typename TestFixture::local_index_type;
+    using global_index_type = typename TestFixture::global_index_type;
+    this->mapping = {this->ref, {1, 0, 2, 2, 0, 1, 1, 2}};
```

Comment:
this mapping is recreated in each test, so it does not need to be in TestFixture